packages/flutter/test/widgets/editable_text_cursor_test.dart

```diff
@@ -791,4 +793,62 @@ void main() {
       matchesGoldenFile('editable_text_test.2.png'),
     );
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
+
+  testWidgets('cursor layout has correct height', (WidgetTester tester) async {
+    final GlobalKey<EditableTextState> editableTextKey = GlobalKey<EditableTextState>();
+
+    String changedValue;
+    final Widget widget = MaterialApp(
+      home: RepaintBoundary(
+        key: const ValueKey<int>(1),
+        child: Column(
+          children: <Widget>[
+            const SizedBox(width: 10, height: 10),
+            EditableText(
+              backgroundCursorColor: Colors.grey,
+              key: editableTextKey,
+              controller: TextEditingController(),
+              focusNode: FocusNode(),
+              style: Typography.material2018(platform: TargetPlatform.iOS).black.subtitle1,
+              cursorColor: Colors.blue,
+              selectionControls: materialTextSelectionControls,
+              keyboardType: TextInputType.text,
+              onChanged: (String value) {
+                changedValue = value;
+              },
+              cursorWidth: 15.0,
+              cursorHeight: 30.0,
+            ),
+          ],
+        ),
+      ),
+    );
+    await tester.pumpWidget(widget);
+
+    // Populate a fake clipboard.
+    const String clipboardContent = 'Hello world!';
+    SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+      if (methodCall.method == 'Clipboard.getData')
+        return const <String, dynamic>{'text': clipboardContent};
+      return null;
+    });
+
+    // Long-press to bring up the text editing controls.
+    final Finder textFinder = find.byKey(editableTextKey);
+    await tester.longPress(textFinder);
+    tester.state<EditableTextState>(textFinder).showToolbar();
+    await tester.pumpAndSettle();
+
+    await tester.tap(find.text('Paste'));
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.pump(const Duration(milliseconds: 500));
```

Comment:
Strange, is the `pump` without a duration needed as well?  And what if you combine the two with durations into one with `Duration(seconds: 1)`?

There are [other tests](https://github.com/flutter/flutter/blob/f98a00c7506a0a4a3a53372e574a048a884ca03b/packages/flutter/test/material/text_field_test.dart#L1664-L1666) that do this sort of thing, though not testing the cursor size.

Anyway I think the main thing is to just make sure it's clear to someone else reading this code why the pumps are there, so maybe just add a comment.

Reply:
It wasn't, I simplified this test and others doing the same thing, and added a comment